plugins/catalog/src/alpha/filter/parseFilterExpression.ts

```diff
@@ -106,7 +114,11 @@ export function splitFilterExpression(
   const result = new Array<{ key: string; parameters: string[] }>();
 
   for (const word of words) {
-    const match = word.match(/^([^:]+):(.+)$/);
+    let match = word.match(/^([^:]+):(.+)$/);
```

Comment:
```suggestion
    let match = word.match(/^(not:)?([^:]+):(.+)$/);
```

how about simply optionally matching on it directly? then checking if `match[1]` is nullish or not. Would simplify things a ton